### PR TITLE
Add SSH_USER env variable, abstracted cleaning tasks to run both in pre-workload and remove-workload

### PR DESCRIPTION
**Pull request by kaovilai**
Adds new environment variable for the workshop.

SSH_USER environment variable now exported to workshop environment such that the guide can list correct username to for user to SSH within the terminal.

There were also Project ownership issues that were fixed by using `kind: ProjectRequest`

Resolved ansible 2.8 dependency by using `k8s_facts` instead of `k8s`'s `wait`

##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME
Role: ocp4-workload-workshop-admin-storage

##### ADDITIONAL INFORMATION
Some of the solutions included using k8s module wait feature which required ansible 2.8 and could be utilized in the future

### CODE REVIEW

File: ansible/roles/ocp4-workload-workshop-admin-storage/tasks/clean-admin.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- name: remove the labguide project
+  command: oc delete project labguide
+  ignore_errors: yes
+
+- name: wait for project to be deleted before continuing
+  command: oc get project labguide
+  register: result
+  retries: 30
+  failed_when: result.stderr.find('Error from server (NotFound)') != 0
+  until: result.stderr.find('Error from server (NotFound)') == 0
```
Comment (fridim) [inline ansible/roles/ocp4-workload-workshop-admin-storage/tasks/clean-admin.yml]: I think you can use

```shell
oc delete project labguide --wait=true
```

That way you don't have to "wait for project to be deleted before continuing"

**Comment by kaovilai** [comment]
FYI, I plan to rollback the idea of using command and shell for the most part. Was told that we would try to not make any changes. But the rest of the changes remain. Changing to WIP

File: ansible/roles/ocp4-workload-workshop-admin-storage/tasks/clean-admin.yml (hunk 1 above)
Comment (kaovilai) [reply ansible/roles/ocp4-workload-workshop-admin-storage/tasks/clean-admin.yml]: @fridim What about using `wait` in `k8s`?

How do I ensure ansible installed is v2.8.*?

Comment (kaovilai) [reply ansible/roles/ocp4-workload-workshop-admin-storage/tasks/clean-admin.yml]: I had
```
yum: name=ansible-2.8.*
```
but that didn't seem to help on first run on the task right after the "upgrade". Or maybe I needed to add wait?

File: ansible/roles/ocp4-workload-workshop-admin-storage/readme.adoc [hunk 2]
```diff
@@ -31,22 +40,16 @@
...
 # a TARGET_HOST is specified in the command line, without using an inventory file
-ansible-playbook -i ${TARGET_HOST}, ./configs/ocp-workloads/ocp-workload.yml \
-    -e"ansible_ssh_private_key_file=/home/lab-user/.ssh/id_rsa" \
-    -e"ansible_user=lab-user" \
-    -e"ocp_username=${OCP_USERNAME}" \
-    -e"ocp_workload=${WORKLOAD}" \
+ansible-playbook -i $(dig @resolver1.opendns.com A myip.opendns.com +short), ./configs/ocp-workloads/ocp-workload.yml \
+    -e"ansible_ssh_private_key_file=/home/$(whoami)/.ssh/id_rsa" \
+    -e"ansible_user=$(whoami)" \
+    -e"ocp_workload=ocp4-workload-workshop-admin-storage" \
```
Comment (fridim) [inline ansible/roles/ocp4-workload-workshop-admin-storage/readme.adoc]: can you remove this from the PR ?
the `dig` part looks very specific to your laptop

File: ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml [hunk 3]
```diff
@@ -1,6 +1,41 @@
 ---
 # Implement your Pre Workload deployment tasks here
 
+- name: Cleaning admin
+  include_tasks:
+    file: ./clean-admin.yml
+
+- name: extract the sandbox id
+  shell: "echo {{ target_host }} | cut -d. -f3"
+  register: sandbox_id
```
Comment (fridim) [inline ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml]: I don't see where you use `sandbox_id` in the rest of the workload. Maybe you can remove that part ?
Also, what happen if the workload is not loaded in a sandbox account but in a standard account ?


We can pass the `sandbox_account` to agnosticd when deploying, that way you won't have to guess is using `target_host` or dns names.

If you tell me which catalog item is using this we can make sure `sandbox_account` is defined.

File: ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml [hunk 4]
```diff
@@ -1,6 +1,41 @@
...
+  shell: echo https://$(oc get route console -o jsonpath='{.spec.host}' -n openshift-console)
+  register: master_url
+
+- name: extract route_subdomain
+  command: oc get ingresses.config.openshift.io cluster -o jsonpath='{.spec.domain}'
+  register: route_subdomain
+
+- name: extract guid
+  shell: echo $(oc whoami --show-server | awk -F '[-.]' '{ print $3 }')
+  register: guid
```
Comment (fridim) [inline ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml]: Do not use the `guid` variable.
This variable is *always* passed as extra vars and will override your register here.

You don't need to extract it.

File: ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml [hunk 5]
```diff
@@ -1,6 +1,41 @@
...
+- name: extract ocp_username
+  command: oc whoami
+  register: ocp_username
+
+- name: cat the kubeadmin-password
+  command: "cat /home/{{ ansible_user }}/cluster-{{ guid.stdout }}/auth/kubeadmin-password"
+  register: kubeadmin_password
+
+- name: extract bastion_fqdn
+  shell: echo "bastion.$(oc whoami --show-server | awk -F '[-:]' '{ print $3 }')"
```
Comment (fridim) [inline ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml]: please prefer

```
bastion.{{ guid }}{{ subdomain_base_suffix }}
```

File: ansible/roles/ocp4-workload-workshop-admin-storage/readme.adoc (hunk 2 above)
Comment (kaovilai) [reply ansible/roles/ocp4-workload-workshop-admin-storage/readme.adoc]: It's basically target_host. Maybe this is extreme but I was told to minimize variables for export.
Before it was ${TARGET_HOST}

File: ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml (hunk 3 above)
Comment (kaovilai) [reply ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml]: This one
image

File: ansible/roles/ocp4-workload-workshop-admin-storage/readme.adoc [hunk 6]
```diff
@@ -31,20 +40,17 @@
...
-    -e"ansible_user=lab-user" \
-    -e"ocp_username=${OCP_USERNAME}" \
-    -e"ocp_workload=${WORKLOAD}" \
+    -e"ansible_ssh_private_key_file=/home/$(whoami)/.ssh/id_rsa" \
+    -e"ansible_user=$(whoami)" \
+    -e"guid=$(oc whoami --show-server | awk -F '[-.]' '{ print $3 }')" \
+    -e"subdomain_base_suffix=bastion.$(oc whoami --show-server | awk -F '[-:]' '{ print $3 }')" \
+    -e"ocp_workload=ocp4-workload-workshop-admin-storage" \
     -e"silent=False" \
-    -e"guid=${GUID}" \
```
Comment (fridim) [inline ansible/roles/ocp4-workload-workshop-admin-storage/readme.adoc]: you need to keep GUID

File: ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml [hunk 7]
```diff
@@ -1,6 +1,33 @@
...
+  include_tasks:
+    file: ./clean-admin.yml
+
+- name: extract api_url
+  command: oc whoami --show-server
+  register: api_url
+
+- name: extract master_url
+  shell: echo https://$(oc get route console -o jsonpath='{.spec.host}' -n openshift-console)
+  register: master_url
```
Comment (fridim) [inline ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml]:
```yaml
- name: extract master_url
  command: oc get route console -o jsonpath='{.spec.host}' -n openshift-console
  register: master_url_r

- set_fact:
    master_url: "https://{{ master_url_r.stdout | trim }}"
```

Comment (fridim) [reply ansible/roles/ocp4-workload-workshop-admin-storage/tasks/pre_workload.yml]: maybe `k8s_facts` can work here too?

**Pull request merged**